compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/events/DisposableEventListener.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.events
+
+import org.w3c.dom.events.Event
+import org.w3c.dom.events.EventTarget
+
+internal fun interface DisposableEventListener {
+    fun dispose()
+}
+
+internal fun EventTarget.addDisposableEvent(eventName: String, handler: (Event) -> Unit): DisposableEventListener {
```

Comment:
Shouldn't this and `DisposableEventListener` be `private`?

Reply:
I actually agree, lesser is better, in theory one can use this separately but in practice it's almost always a bunch of events (and even in case of one event we always can use EventTargetListener)

pushed this changes